woodwork/type_sys/inference_functions.py

```diff
@@ -94,3 +94,23 @@ def email_address_func(series: pd.Series) -> bool:
     matches = series_match_method(pat=regex)
 
     return matches.sum() == matches.count()
+
+
+def url_func(series: pd.Series) -> bool:
+    regex = ww.config.get_option('url_inference_regex')
+
+    # Includes a check for object dtypes
+    if not pdtypes.is_string_dtype(series.dtype):
+        return False
+
+    try:
+        series_match_method = series.str.match
+    except (AttributeError, TypeError):
+        # This can happen either when the inferred dtype for a series is not
+        # compatible with the pandas string API (AttributeError) *or* when the
+        # inferred dtype is not compatible with the string API `match` method
+        # (TypeError)
+        return False
+    matches = series_match_method(pat=regex)
+
+    return matches.sum() == matches.count()
```

Comment:
Since this inference func is really just a copy of the email inference func, I wonder if it makes sense to abstract this out into a class.  So something like this:
```python
class InferWithRegex:
    def __init__(self, get_regex: Callable[[], str]):
        self.get_regex = get_regex

    def __call__(self, series: pd.Series) -> bool:
        regex = self.get_regex()

        # Includes a check for object dtypes
        if not pdtypes.is_string_dtype(series.dtype):
            return False

        try:
            series_match_method = series.str.match
        except (AttributeError, TypeError):
            # This can happen either when the inferred dtype for a series is not
            # compatible with the pandas string API (AttributeError) *or* when the
            # inferred dtype is not compatible with the string API `match` method
            # (TypeError)
            return False
        matches = series_match_method(pat=regex)

        return matches.sum() == matches.count()

email_address_func = InferWithRegex(lambda: ww.config.get_option('email_inference_regex'))
url_func = InferWithRegex(lambda: ww.config.get_option('url_inference_regex'))
```
You would probably want to use the lambdas (or some kind of deferred config lookup) to avoid having to worry about what order things are done in.  For example, a user might happen to import the module that these funcs are defined in before they have a chance to override the config setting.

Reply:
It does make more sense to abstract it into a class. I used your suggestion and pushed new changes.